plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/TestElasticsearchMetadata.java

```diff
@@ -13,15 +13,91 @@
+import com.google.common.io.BaseEncoding;
 import io.airlift.slice.Slices;
+import io.trino.plugin.elasticsearch.client.IndexMetadata;
+import io.trino.plugin.elasticsearch.client.TestingElasticsearchClient;
+import io.trino.spi.TrinoException;
+import io.trino.spi.connector.SchemaTableName;
+import io.trino.spi.type.TestingTypeManager;
+import org.intellij.lang.annotations.Language;
 import org.testng.annotations.Test;
 
+import java.util.Map;
 import java.util.Optional;
 
+import static io.trino.type.IpAddressType.IPADDRESS;
+import static java.lang.String.format;
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.assertj.core.api.Assertions.assertThatNoException;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
 import static org.testng.Assert.assertEquals;
 
 public class TestElasticsearchMetadata
 {
+    @Test
+    public void testPassThroughQueryWithEnabledFeature()
+    {
+        @Language("JSON")
+        String query = "{\n" +
+                "    \"size\": 0,\n" +
+                "    \"aggs\" : {\n" +
+                "        \"max_orderkey\" : { \"max\" : { \"field\" : \"orderkey\" } },\n" +
+                "        \"sum_orderkey\" : { \"sum\" : { \"field\" : \"orderkey\" } }\n" +
+                "    }\n" +
+                "}";
+
+        String schemaName = "test_schema";
+        Map<String, IndexMetadata> indices = ImmutableMap.of(
+                schemaName,
+                new IndexMetadata(new IndexMetadata.ObjectType(ImmutableList.of())));
+        ElasticsearchMetadata elasticsearchMetadata = new ElasticsearchMetadata(
+                new TestingTypeManager(ImmutableList.of(IPADDRESS)),
+                new TestingElasticsearchClient(indices),
+                new ElasticsearchConfig()
+                        .setDefaultSchema(schemaName)
+                        .setLegacyPassThroughQueryEnabled(true));
+
+        SchemaTableName schemaTableName = SchemaTableName.schemaTableName(
+                schemaName,
+                format("orders$query:%s", BaseEncoding.base32().encode(query.getBytes(UTF_8))));
+
+        assertThatNoException().isThrownBy(() -> elasticsearchMetadata.getTableHandle(null, schemaTableName));
+    }
+
+    @Test
+    public void testPassThroughQueryWithDisabledFeature()
+    {
+        @Language("JSON")
+        String query = "{\n" +
+                "    \"size\": 0,\n" +
+                "    \"aggs\" : {\n" +
+                "        \"max_orderkey\" : { \"max\" : { \"field\" : \"orderkey\" } },\n" +
+                "        \"sum_orderkey\" : { \"sum\" : { \"field\" : \"orderkey\" } }\n" +
+                "    }\n" +
+                "}";
+
+        String schemaName = "test_schema";
+        Map<String, IndexMetadata> indices = ImmutableMap.of(
+                schemaName,
+                new IndexMetadata(new IndexMetadata.ObjectType(ImmutableList.of())));
+        ElasticsearchMetadata elasticsearchMetadata = new ElasticsearchMetadata(
```

Comment:
We don't do such testing ;) We try to avoid plumbing. Instead please create a new test class that extends abstract test query framewok and create new query runner that query pass through disabled. Then you can simply do `assertQueryFail`